BlueShift-iOS-SDK/BlueShiftDeviceData.m

```diff
@@ -93,8 +93,8 @@ - (NSDictionary *)toDictionary {
         [deviceMutableDictionary setObject:self.deviceType forKey:@"device_type"];
     }
     
-    if (self.deviceToken) {
-        NSString *storedDeviceToken = [[BlueShift sharedInstance] getDeviceToken];
+    NSString *storedDeviceToken = [[BlueShift sharedInstance] getDeviceToken];
+    if (storedDeviceToken) {
```

Comment:
ok, so we are not looking into the live object all?

Reply:
Yes. When a token is received from OS its getting stored in the UserDefaults and this way it will always pick the stored token.

Reply:
ok, so no need to look into the live variable at all. got it.